### PR TITLE
feat: finalize implementation in circular doubly linked list

### PR DESCRIPTION
**Pull request by manojnakp**
Add documentation and prepare for release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,16 +33,16 @@ import (
```
Comment (bingxueshuang) [inline README.md:33]: verify that the example code runs properly on [playground](https://go.dev/play)

File: interface.go [hunk 2]
```diff
@@ -46,7 +46,16 @@ type Interface[T any] interface {
 }
```
Comment (bingxueshuang) [inline interface.go:47]: good way of implementing `Must` wrapper

File: list.go [hunk 3]
```diff
@@ -44,6 +44,9 @@ func (d *Deque[T]) Back() (item T, err error) {
```
Comment (bingxueshuang) [inline list.go:44]: Handle `nil` deque carefully

File: README.md (hunk 1 above)
Comment (manojnakp) [reply README.md:33]: https://go.dev/play/p/fYkdsj9V20U

**Review by bingxueshuang — APPROVED** [review]

**Pull request merged**